sklearn/utils/_metadata_requests.py

```diff
@@ -1024,7 +1026,9 @@ def _serialize(self):
         """
         res = dict()
         if self._self_request:
-            res["$self_request"] = self._self_request._serialize()
+            res["$self_request"] = (
+                self._self_request._serialize()
+            )  # <-- why is `$` part of the key name?
```

Comment:
because it makes it an invalid identifier, which means developers are not gonna use it as a name for a sub-object in routing structure when they add things to `MetadataRouter`, so there will be no conflicts.